src/GafferScene/Constraint.cpp

```diff
@@ -525,58 +516,52 @@ struct UVIndexer
 			}
 		}
 
-		m_uvs = &( IECore::assertedStaticCast< const IECore::V2fVectorData >( uvIt->second.data )->readable() );
+		// NOTE : for vertex and varying interpolation we need to redirect through the primitive indices
 
-		switch( uvIt->second.interpolation )
+		if(
+			( ( *it ).second.interpolation == IECoreScene::PrimitiveVariable::Vertex ) ||
+			( ( *it ).second.interpolation == IECoreScene::PrimitiveVariable::Varying ) )
 		{
-			case IECoreScene::PrimitiveVariable::Interpolation::FaceVarying:
-				if( uvIt->second.indices )
-				{
-					m_indices = &( uvIt->second.indices->readable() );
-				}
-				break;
-			case IECoreScene::PrimitiveVariable::Interpolation::Vertex:
-				m_indices = &( primitive.vertexIds()->readable() );
-				if( uvIt->second.indices )
-				{
-					m_uvIndices = &( uvIt->second.indices->readable() );
-				}
-				break;
-			default:
-				if( throwOnError )
-				{
-					throw IECore::InvalidArgumentException( (
-						boost::format(
-							"Gaffer::Constraint : Primitive variable named \"%s\"  has incorrect interpolation, must be either Vertex or FaceVarying"
-						) % ( uvSet ) ).str() );
-				}
-				else
-				{
-					return;
-				}
-				break;
+			m_indices = &( primitive.vertexIds()->readable() );
+		}
+		else if( ( *it ).second.interpolation != IECoreScene::PrimitiveVariable::FaceVarying )
+		{
+			if( throwOnError )
+			{
+				throw IECore::InvalidArgumentException( (
+					boost::format(
+						"Gaffer::Constraint : Primitive variable named \"%s\" has incorrect interpolation, must be either Vertex, Varying or FaceVarying"
+					) % ( uvSet ) ).str() );
+			}
+			else
+			{
+				return;
+			}
 		}
+
+		// NOTE : build the indexed view.
+
+		m_view = IECoreScene::PrimitiveVariable::IndexedView< Imath::V2f >(
```

Comment:
This doesn't change functionality at all, but I think what you're doing here is just the same stuff that the IndexedView constructor does when passed a variable, and you could just call `IECoreScene::PrimitiveVariable::IndexedView< Imath::V2f >( it->second );` ?